runtime/drivers/snowflake/sql_store.go

```diff
@@ -184,13 +204,47 @@ func (f *fileIterator) Next() ([]string, error) {
 	// write arrow records to parquet file
 	// the following iteration might be memory intensive
 	// since batches are organized as a slice and every batch caches its content
+	batchesLeft := len(f.batches)
+	f.logger.Debug("starting to fetch and process arrow batches",
+		zap.Int("batches", batchesLeft), zap.Int("parallel_fetch_limit", f.parallelFetchLimit))
+
+	// Fetch batches async as it takes most of the time
+	var wg sync.WaitGroup
+	fetchResultChan := make(chan fetchResult, len(f.batches))
+	sem := semaphore.NewWeighted(int64(f.parallelFetchLimit))
 	for _, batch := range f.batches {
-		records, err := batch.Fetch()
-		if err != nil {
-			return nil, err
+		wg.Add(1)
+		go func(b *sf.ArrowBatch) {
+			defer wg.Done()
+			defer sem.Release(1)
+			err := sem.Acquire(f.ctx, 1)
+			if err != nil {
+				fetchResultChan <- fetchResult{Records: nil, Batch: nil, Err: err}
+				return
+			}
```

Comment:
I think releases should be after successful acquire (else might panic)

Reply:
removed the use of a semaphore